enterprise/neo4j-harness-enterprise/src/main/java/org/neo4j/harness/CausalClusterInProcessRunner.java

```diff
@@ -0,0 +1,229 @@
+                builder.withConfig( CausalClusteringSettings.raft_listen_address.name(), specifyPortOnly( raftPort ) );
+
+                builder.withConfig( CausalClusteringSettings.expected_core_cluster_size.name(), String.valueOf( nCores ) );
+                builder.withConfig( CausalClusteringSettings.server_tags.name(), "core," + "core" + coreId );
+                configureConnectors( boltPort, httpPort, builder );
+
+                builder.withConfig( ServerSettings.jmx_module_enabled.name(), Settings.FALSE );
+
+                int finalCoreId = coreId;
+                Thread coreThread = new Thread( () ->
+                {
+                    coreControls.add( builder.newServer() );
+                    log.info( "Core " + finalCoreId + " started." );
+                } );
+                coreThreads.add( coreThread );
+                coreThread.start();
+            }
+
+            for ( Thread coreThread : coreThreads )
+            {
+                coreThread.join();
+            }
+
+            for ( int replicaId = 0; replicaId < nCores; replicaId++ )
+            {
+                int txPort = 56500 + replicaId;
+                int boltPort = 58500 + replicaId;
+                int httpPort = 59500 + replicaId;
+
+                String homeDir = "replica-" + replicaId;
+                TestServerBuilder builder = new EnterpriseInProcessServerBuilder( clusterPath.toFile(), homeDir );
+
+                String homePath = Paths.get( clusterPath.toString(), homeDir ).toAbsolutePath().toString();
+                builder.withConfig( GraphDatabaseSettings.neo4j_home.name(), homePath );
+                builder.withConfig( GraphDatabaseSettings.pagecache_memory.name(), "8m" );
+
+                builder.withConfig( ClusterSettings.mode.name(), ClusterSettings.Mode.READ_REPLICA.name() );
+                builder.withConfig( CausalClusteringSettings.initial_discovery_members.name(), String.join( ",", initialMembers ) );
+                builder.withConfig( CausalClusteringSettings.transaction_listen_address.name(), specifyPortOnly( txPort ) );
+
+                builder.withConfig( CausalClusteringSettings.server_tags.name(), "replica," + "replica" + replicaId );
+                configureConnectors( boltPort, httpPort, builder );
+
+                builder.withConfig( ServerSettings.jmx_module_enabled.name(), Settings.FALSE );
+
+                int finalReplicaId = replicaId;
+                Thread replicaThread = new Thread( () ->
+                {
+                    replicaControls.add( builder.newServer() );
+                    log.info( "Read replica " + finalReplicaId + " started." );
+                } );
+                replicaThreads.add( replicaThread );
+                replicaThread.start();
+            }
+
+            for ( Thread replicaThread : replicaThreads )
+            {
+                replicaThread.join();
+            }
+        }
+
+        private static String specifyPortOnly( int port )
```

Comment:
@jimwebber Well, I specifically factored it out to increase understandability. The syntax of :port is not really commonplace I believe and would look something like ":" + port inline, so I found this better. This at least tells you something about what that syntax means. It would of course better if this was a supported setting type with helper functions.